test/e2e/operator_test.go

```diff
@@ -70,172 +76,175 @@ var _ = Describe("Operator API", func() {
-		o := &operatorsv1.Operator{}
-		o.SetName(genName("o-"))
-
-		Consistently(o).ShouldNot(ContainCopiedCSVReferences())
-
-		Eventually(func() error {
-			return client.Create(clientCtx, o)
-		}).Should(Succeed())
+	Context("when an Operator resource can select its components by label", func() {
+		var (
+			o *operatorsv1.Operator
+		)
+		BeforeEach(func() {
+			o = &operatorsv1.Operator{}
+			o.SetName(genName("o-"))
 
-		defer func() {
 			Eventually(func() error {
-				err := client.Delete(clientCtx, o)
-				if apierrors.IsNotFound(err) {
-					return nil
-				}
-
-				return err
+				return client.Create(clientCtx, o)
+			}).Should(Succeed())
+		})
+		AfterEach(func() {
+			Eventually(func() error {
+				return controllerclient.IgnoreNotFound(client.Delete(clientCtx, o))
 			}).Should(Succeed())
-		}()
+		})
 
-		By("eventually having a status that contains its component label selector")
-		w, err := operatorClient.Watch(clientCtx, listOpts)
-		Expect(err).ToNot(HaveOccurred())
-		defer w.Stop()
+		It("should not contain copied csv status references", func() {
+			Consistently(o).ShouldNot(ContainCopiedCSVReferences())
+		})
 
-		deadline, cancel := context.WithTimeout(clientCtx, 1*time.Minute)
-		defer cancel()
+		It("should surface referenced components in its status", func() {
+			By("eventually having a status that contains its component label selector")
+			w, err := operatorClient.Watch(clientCtx, listOpts)
+			Expect(err).To(BeNil())
+			defer w.Stop()
 
-		expectedKey := "operators.coreos.com/" + o.GetName()
-		awaitPredicates(deadline, w, operatorPredicate(func(op *operatorsv1.Operator) bool {
-			if op.Status.Components == nil || op.Status.Components.LabelSelector == nil {
-				return false
-			}
+			deadline, cancel := context.WithTimeout(clientCtx, 1*time.Minute)
+			defer cancel()
 
-			for _, requirement := range op.Status.Components.LabelSelector.MatchExpressions {
-				if requirement.Key == expectedKey && requirement.Operator == metav1.LabelSelectorOpExists {
-					return true
+			expectedKey := "operators.coreos.com/" + o.GetName()
+			awaitPredicates(deadline, w, operatorPredicate(func(op *operatorsv1.Operator) bool {
+				if op.Status.Components == nil || op.Status.Components.LabelSelector == nil {
+					return false
 				}
-			}
 
-			return false
-		}))
-		defer w.Stop()
+				for _, requirement := range op.Status.Components.LabelSelector.MatchExpressions {
+					if requirement.Key == expectedKey && requirement.Operator == metav1.LabelSelectorOpExists {
+						return true
+					}
+				}
 
-		// Create namespaces ns-a and ns-b
-		nsA := &corev1.Namespace{}
-		nsA.SetName(genName("ns-a-"))
-		nsB := &corev1.Namespace{}
-		nsB.SetName(genName("ns-b-"))
+				return false
+			}))
+			defer w.Stop()
 
-		for _, ns := range []*corev1.Namespace{nsA, nsB} {
-			Eventually(func() error {
-				return client.Create(clientCtx, ns)
-			}).Should(Succeed())
+			// Create namespaces ns-a and ns-b
```

Comment:
nit: these comments are just so unnecessary

Reply:
That diff is misleading - I was simply indenting that existing codepaths, but agreed, I don't see a ton of value in this style of comment either.

Reply:
I ended up removing them entirely, or replacing them with `By(...)` statements in the latest force push.